libeos-updater-util/flatpak.c

```diff
@@ -1673,20 +1689,23 @@ euu_lookup_flatpak_remote_for_collection_id (FlatpakInstallation  *installation,
 
   remotes = ostree_repo_remote_list (repo, NULL);
 
-  for (iter = remotes; *iter != NULL; ++iter)
+  if (remotes)
     {
-      g_autofree gchar *remote_collection_id = NULL;
-
-      if (!ostree_repo_get_remote_option (repo,
-                                          *iter,
-                                          "collection-id",
-                                          NULL,
-                                          &remote_collection_id,
-                                          error))
-        return NULL;
-
-      if (g_strcmp0 (remote_collection_id, collection_id) == 0)
-        return g_strdup (*iter);
+      for (iter = remotes; *iter != NULL; ++iter)
```

Comment:
Nitpick: You can avoid adding an unnecessary level of indentation here by combining:
```
if (A)
  {
    for (S; B; I)
      …
  }
```
as
```
for (S; A && B; I)
  …
```

No need to change that here now though.

Reply:
Doesn't that check the list head to be NULL on every iteration though? Or does the compiler optimize it away?

Reply:
The compiler might optimise it away (depends how good its analysis of the invariants on `A` in the loop body are). Even if it doesn’t, would you be able to time the extra time it takes to do that check on each of ~10 configured remotes? :-)